### PR TITLE
Lets you clear a single recipe from dispensers

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
As title. Hit the delete recipes button, choose 'only one', and then press the recipe you wanna clear.
Also fixes a bug with dispenser fumble checks.

## Why It's Good For The Game
Being able to clear a specific recipe is better than having to clear all recipes if you want to remove one and then readd all the others.

## Changelog
:cl:
add: You can choose to only remove a single chem recipe instead of all of them when using a dispenser.
fix: You can't start fiddling with a chem dispenser while you're busy doing something else.
/:cl:

### CODE REVIEW

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -249,8 +258,11 @@
 		if("clear_recipes")
 			if(!is_operational())
 				return
-			var/yesno = tgui_alert(usr, "Clear all recipes?", null, list("Yes","No"))
-			if(yesno == "Yes")
+			var/confirm_clear = tgui_alert(usr, "Clear all recipes?", null, list("Yes","No", "Only one"))
```
Comment (BraveMole) [inline code/modules/reagents/machinery/chem_dispenser.dm:261]: use a switch

**Review by BraveMole — CHANGES_REQUESTED** [review]
Can't you add a button to directly clear the specific recipe, rather than this two steps thingy?

**Comment by QualityVan** [comment]
> 
> 
> Can't you add a button to directly clear the specific recipe, rather than this two steps thingy?

If I do it this way I don't have to mess with tgui stuff. Also, I'm not sure about single-stage clearing. Would each recipe have its own remove button? You could clear by having the button let you input a recipe name, but that just seems less convenient than a two-button option.

**Comment by SplinterGP** [comment]
I think this delete mode is better because having a delete button on the side of EACH recipe  is a bit annoying

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -222,6 +227,13 @@
 		if("dispense_recipe")
 			if(!is_operational() || QDELETED(cell))
 				return
+			if(clearing_recipe)
+				var/actually_clear = tgui_alert(usr, "Clear recipe [params["recipe"]]?", null, list("Yes","No"))
+				if(actually_clear == "Yes")
```
Comment (BraveMole) [inline code/modules/reagents/machinery/chem_dispenser.dm:232]:
```suggestion
				if(tgui_alert(usr, "Clear recipe [params["recipe"]]?", null, list("Yes","No")  == "Yes"))
```

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 3]
```diff
@@ -222,6 +227,13 @@
...
+					usr.client.prefs.chem_macros.Remove(params["recipe"])
+					usr.client.prefs.save_preferences()
+				clearing_recipe = FALSE
```
Comment (BraveMole) [inline code/modules/reagents/machinery/chem_dispenser.dm:235]: eh, you have a clear recipe button to remove that already

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 4]
```diff
@@ -249,14 +261,21 @@
...
 				return
-			var/yesno = tgui_alert(usr, "Clear all recipes?", null, list("Yes","No"))
-			if(yesno == "Yes")
-				usr.client.prefs.chem_macros = list()
-				usr.client.prefs.save_preferences()
+			if(clearing_recipe)
+				clearing_recipe = FALSE
+				return TRUE
+			var/confirm_clear = tgui_alert(usr, "Clear all recipes?", null, list("Yes","No", "Only one"))
+			switch(confirm_clear)
```
Comment (BraveMole) [inline code/modules/reagents/machinery/chem_dispenser.dm:268]: ditto, useless var

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**